File: Codeforces/global22/A.cpp

#include <iostream>
#include <vector>
#include <algorithm>

using namespace::std;
typedef unsigned long long ull;

ull solve(const vector<ull> &A, const vector<ull> &B)
{
	int aa = 0;
	int bb = 0;

	ull ans = 0;
	if (A.size() > 0)
	{
		ans = A.back();
		aa = 1;
	}
	for (ull i=0; i < max(A.size(), B.size()) + 1; ++i)
	{
		if (i < B.size())
		{
			if (aa)
				ans += B.at(i) * 2;
			else
				ans += B.at(i);
			bb = 1;
		} else
			bb = 0;


		if (A.size() > 0 and i < A.size() - 1)
		{
			if (bb)
				ans += A.at(i) * 2;
			else
				ans += A.at(i);
			aa = 1;
		} else
			aa = 0;
	}
	return ans;
}

int main()
{
	int t;
	cin >> t;
	while (t--)
	{
		ull n;
		cin >> n;
		vector<ull> a(n);
		vector<ull> b(n);

		for (ull i=0; i < n; ++i)
			cin >> a.at(i);
		for (ull i=0; i < n; ++i)
			cin >> b.at(i);

		vector<ull> A;
		vector<ull> B;
		A.reserve(n);
		B.reserve(n);

		for (ull i=0; i < n; ++i)
		{
			if (a.at(i) == 0)
				A.push_back(b.at(i));
		}
		for (ull i=0; i < n; ++i)
		{
			if (a.at(i) == 1)
				B.push_back(b.at(i));
		}


		std::sort(A.begin(), A.end(), [](ull aa, ull bb){ return aa > bb; });
		std::sort(B.begin(), B.end(), [](ull aa, ull bb){ return aa > bb; });
		cout << max(solve(A, B), solve(B, A)) << endl;
	}
}

File: Codeforces/global22/A.py

def inp():
    return list(map(int, input().split()))

def __s(A, B):
    if i >= len(A) and i >= len(B):
        pass
    if i < len(A) and i < len(B):
        a1 += A[i] * 2 + B[i] * 2
    elif i < len(A):
        a1 += A[i]
    elif i < len(B):
       a1 += B[i]

    if len(A) > 0:
        a1 -= A[-1]
    
    return a1

def solve(A, B):
    aa = False
    bb = False
    ans = 0

    if len(A) > 0:
        ans = A[-1]
        aa = True
    for i in range(max(len(A), len(B)) + 1):
        if i < len(B):
            if aa:
                ans += B[i] * 2
            else:
                ans += B[i]
            bb = True
        else:
            bb = False

        if i < len(A) - 1:
            if bb:
                ans += A[i] * 2
            else:
                ans += A[i]
            aa = True
        else:
            aa = False

    return ans

for _ in range(int(input())):
    n = int(input())
    a = inp()
    b = inp()

    A = [b[i] for i in range(n) if a[i] == 0]
    B = [b[i] for i in range(n) if a[i] == 1]

    A.sort(reverse=True)
    B.sort(reverse=True)

    print(max(solve(A, B), solve(B, A)))

